### PR TITLE
[WIP]feature: Refactoring the Doku component to reuse it for voucher variants and built the vaoucher UI and action component

### PR DESCRIPTION
**Pull request by mohammedDehairy**
feature: Refactoring the Doku component to reuse it for voucher variants and built the vaoucher UI and action component

### CODE REVIEW

File: AdyenActions/Actions/Action.swift [hunk 1]
```diff
@@ -59,6 +64,7 @@ public enum Action: Decodable {
         case sdk
         case qrCode
         case await
+            case voucher
```
Comment (descorp) [inline AdyenActions/Actions/Action.swift:67]: what happened with indent here? :D

File: AdyenActions/Actions/VoucherAction.swift [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+/// Describes an action in which a voucher is presented to the shopper.
+public struct DokuVoucherAction: Decodable {
+
+    /// The `paymentMethodType` for which the voucher is presented.
+    public let paymentMethodType: VoucherPaymentMethod
+
+    /// The initial  amount.
+    public let initialAmount: Payment.Amount
```
Comment (descorp) [inline AdyenActions/Actions/VoucherAction.swift:49]: Nice :)

File: AdyenActions/UI/View Controllers/Voucher/VoucherSeparatorView.swift [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import Adyen
+import CoreGraphics
+import QuartzCore
+import UIKit
+
+extension VoucherSeparatorView {
+    internal struct Model {
+        internal var separatorTitle: String? = "Payment code"
+
+        internal var separatorTitlestyle = TextStyle(font: .systemFont(ofSize: 13),
```
Comment (descorp) [inline AdyenActions/UI/View Controllers/Voucher/VoucherSeparatorView.swift:16]: separatorTitlestyle - > separatorTitleStyle

File: Demo/Configuration.swift [hunk 4]
```diff
@@ -28,7 +28,7 @@ internal enum Configuration {
     
     static let shopperReference = "iOS Checkout Shopper"
 
-    static let merchantAccount = "TestMerchantCheckout"
+    static let merchantAccount = "TestMerchant"
```
Comment (descorp) [inline Demo/Configuration.swift:31]: Do we need this?

File: Demo/UIKit/Info.plist [hunk 5]
```diff
@@ -79,6 +79,8 @@
 		<key>NSAllowsLocalNetworking</key>
 		<true/>
 	</dict>
+	<key>NSPhotoLibraryAddUsageDescription</key>
+	<string>Save vouchers to your photos library</string>
```
Comment (descorp) [inline Demo/UIKit/Info.plist:83]: We better not forget to mention this in our docs :D

Comment (mohammedDehairy) [reply Demo/UIKit/Info.plist:83]: Yeah, true, I'll add it to the readme file.

Comment (mohammedDehairy) [reply Demo/UIKit/Info.plist:83]: added a paragraph to the ReadME file.

File: Demo/Configuration.swift (hunk 4 above)
Comment (mohammedDehairy) [reply Demo/Configuration.swift]: reverted

File: AdyenActions/UI/View Controllers/Voucher/VoucherSeparatorView.swift (hunk 3 above)
Comment (mohammedDehairy) [reply AdyenActions/UI/View Controllers/Voucher/VoucherSeparatorView.swift]: 👍

File: AdyenActions/Actions/Action.swift (hunk 1 above)
Comment (mohammedDehairy) [reply AdyenActions/Actions/Action.swift]: swiftFormat is behaving very weirdly.

**Review by descorp — APPROVED** [review]

**Pull request merged**